lib/assigner/test/assigner_test.cpp

```diff
@@ -164,24 +190,127 @@ TEST_F(AssignerTest, conversions_uint64_to_zkevm_word)
-    // load data to the lase 64 bits
-    tmp.load_partial_data(reinterpret_cast<const uint8_t*>(&number), 8, 0);
-    uint64_t number_result;
-    // store back to the uint64_t variable
-    tmp.store<uint64_t>(reinterpret_cast<uint8_t*>(&number_result));
-    // check if same
-    EXPECT_EQ(number_result, number);
+    constexpr size_t word_size = 8;
+    constexpr auto num_full_words = bytes.size() / word_size;
+    constexpr auto num_partial_bytes = bytes.size() % word_size;
+    auto data = bytes.data();
+
+    // Load top partial word.
+    if constexpr (num_partial_bytes != 0)
+    {
+        tmp.load_partial_data(data, num_partial_bytes, num_full_words);
+        data += num_partial_bytes;
+    }
+
+    // Load full words.
+    for (size_t i = 0; i < num_full_words; ++i)
+    {
+        tmp.load_partial_data(data, word_size, num_full_words - 1 - i);
+        data += word_size;
+    }
+    ASSERT_EQ(bytes_to_string(bytes.data(), bytes.size()), intx::to_string(tmp.to_value_type(), 16));
+}
+
+TEST_F(AssignerTest, field_bitwise_and) {
+    using intx::operator""_u256;
+    auto bits_set_zkevm_word = [](std::initializer_list<unsigned> bits) {
+        nil::blueprint::zkevm_word<BlueprintFieldType> tmp(0);
+        for (unsigned bit_number : bits)
+        {
+            tmp = tmp + nil::blueprint::zkevm_word<BlueprintFieldType>(1_u256 << bit_number);
+        }
+        return tmp;
+    };
+    auto bits_set_integral = [](std::initializer_list<unsigned> bits) {
+        BlueprintFieldType::integral_type tmp = 0;
+        for (unsigned bit_number : bits)
+        {
+            boost::multiprecision::bit_set(tmp, bit_number);
+        }
+        return tmp;
+    };
+    auto bits_test_integral = [&](BlueprintFieldType::integral_type val,
+                                  std::unordered_set<unsigned> bits) {
+        for (unsigned bit_number = 0; bit_number < BlueprintFieldType::number_bits; ++bit_number)
+        {
+            EXPECT_EQ(boost::multiprecision::bit_test(val, bit_number), bits.contains(bit_number));
+        }
+    };
+    std::initializer_list<unsigned> init_bits = {1, 12, 42, 77, 136, 201, 222};
+    nil::blueprint::zkevm_word<BlueprintFieldType> init = bits_set_zkevm_word(init_bits);
+
+    std::initializer_list<unsigned> conjunction_bits = {4, 12, 77, 165, 222};
+    BlueprintFieldType::integral_type op = bits_set_integral(conjunction_bits);
+
+    BlueprintFieldType::integral_type res = init & op;
+
+    std::unordered_set<unsigned> res_bit_set;
+    std::set_intersection(init_bits.begin(), init_bits.end(), conjunction_bits.begin(),
+        conjunction_bits.end(), std::inserter(res_bit_set, res_bit_set.begin()));
+    bits_test_integral(res, res_bit_set);
+}
+
+TEST_F(AssignerTest, w_hi_lo)
+{
+    using intx::operator""_u256;
+    auto hi = 0x12345678901234567890_u256;
+    auto lo = 0x98765432109876543210_u256;
+    nil::blueprint::zkevm_word<BlueprintFieldType> tmp((hi << 128) + lo);
```

Comment:
result of + is field, right?
if so, some overflow possible.
May be better:
```
hi = zkevm_word<BlueprintFieldType>(hi)
hi <<= 128
lo = zkevm_word<BlueprintFieldType>(lo)
tmp = hi + lo
```

Reply:
No, actually the result is `intx::uint256`. I agree that this is not obvious, so I can replace `auto` with explicit types. But there should be no overflow possible here, the type is big enough.

Reply:
sorry, it's `_u256`, no questions then